src/helpers/blood-pressure-data-providers/survey-blood-pressure-data-provider.ts

```diff
@@ -23,8 +23,13 @@ export default async function (props: SurveyBloodPressureDataParameters): Promis
             let resultIds = [...new Set(answers.map(a => a.surveyResultID))];
             resultIds.forEach((resultId) => {
                 var resultsForSubmission = answers.filter(a => a.surveyResultID == resultId);
-                var bpLogDateResults = resultsForSubmission.find(r => r.resultIdentifier == props.dateResultIdentifier);
-                var bpLogDate = bpLogDateResults && bpLogDateResults.answers ? bpLogDateResults.answers[0] : "";
+                var bpLogDate = "";
+                if (props.dateResultIdentifier){
+                    var bpLogDateResults = resultsForSubmission.find(r => r.resultIdentifier == props.dateResultIdentifier);
+                    bpLogDate = bpLogDateResults && bpLogDateResults.answers ? bpLogDateResults.answers[0] : "";
```

Comment:
If we don't get a date result in the survey, should we use also use the the same fallback to the survey date?

Reply:
@brandoncharnesky See line 38. If a survey data result id is supplied then it is a required field in terms of the BP Entry. The entry will not be shown if the date/sys/dias are all present